specification/marketplacecatalog/data-plane/Products/preview/2022-09-01-preview/products.json

```diff
@@ -0,0 +1,931 @@
+          },
+          "404": {
+            "x-nullable": false,
+            "description": "In case product does not exist, or does not exist in given market",
+            "schema": {
+              "type": "string"
+            }
+          },
+          "400": {
+            "x-nullable": false,
+            "description": "In case request is malformed",
+            "schema": {
+              "type": "string"
+            }
+          }
+        }
+      }
+    },
+    "/products": {
+      "get": {
+        "tags": [
+          "ListingPublicProducts"
+        ],
+        "summary": "Returns a list of publicly available products",
+        "x-ms-examples": {
+          "Product_Get": {
+            "$ref": "./examples/ListingPublicProducts_Get.json"
+          }
+        },
+        "operationId": "PublicProducts_Get",
+        "parameters": [
+          {
+            "type": "string",
+            "name": "x-ms-app",
+            "x-originalName": "callingApp",
+            "in": "header",
+            "required": true,
+            "description": "The Application named property has the name of the client application that makes the request, used for tracing. E.g. 'appsource'",
+            "x-nullable": true
+          },
+          {
+            "type": "string",
+            "name": "storefront",
+            "in": "query",
+            "description": "Relevant storefront, possible values - 'azure, amp, appsource'. Default='azure'",
+            "default": "azure",
+            "x-nullable": true
+          },
+          {
+            "type": "string",
+            "name": "api-version",
+            "x-originalName": "apiVersion",
+            "in": "query",
+            "required": true,
+            "description": "Version of the REST API used for the request. ",
+            "default": "2022-09-01-preview",
+            "x-nullable": true
+          },
+          {
+            "type": "string",
+            "name": "language",
+            "in": "query",
+            "description": "Language to search, ISO 639-1 two-letter code, possible values - 'en,cs,de,es,fr,hu,it,ja,ko,nl,pl,pt-br,pt-pt,ru,sv,tr,zh-hans,zh-hant'",
+            "default": "en",
+            "x-nullable": true
+          },
+          {
+            "type": "string",
+            "name": "market",
+            "in": "query",
+            "description": "Product sold market, possible values at https://docs.microsoft.com/en-us/azure/marketplace/marketplace-geo-availability-currencies, such as ['US']",
+            "x-nullable": true
+          },
+          {
+            "type": "array",
+            "name": "hideKeys",
+            "in": "query",
+            "description": "Add hide key to retrieve preview products",
+            "collectionFormat": "multi",
+            "x-nullable": true,
+            "items": {
+              "type": "string"
+            }
+          },
+          {
+            "type": "string",
+            "name": "$select",
```

Comment:
Don’t use the “$” prefix character on $select / $filter / $expand / $orderby. The [Azure API Guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md#query-options) require query parameters to be camel case and specifically prohibit the “$” prefix on parameters borrowed from Data.